Mathlib/Data/Polynomial/Degree/Lemmas.lean

```diff
@@ -315,6 +315,15 @@ theorem natDegree_map_eq_iff {f : R →+* S} {p : Polynomial R} :
   simp [h, h2, h3] -- simp doesn't rewrite in the hypothesis for some reason
   tauto
 
+theorem natDegree_pos_of_nz_nextCoeff (h : p.nextCoeff ≠ 0) : 0 < p.natDegree := by
+  rw [nextCoeff] at h
+  by_cases hpz : p.natDegree = 0
+  · simp_all only [ne_eq, zero_le, ite_true, not_true_eq_false]
+  · apply Nat.zero_lt_of_ne_zero hpz
+
+theorem ne_zero_of_nz_nextCoeff (h : p.nextCoeff ≠ 0) : p ≠ 0 :=
+  ne_zero_of_natDegree_gt (natDegree_pos_of_nz_nextCoeff h)
```

Comment:
That's just a trivial restatement of `nextCoeff 0 = 0`, right? Do you really need this?

Reply:
Yup, oops

Reply:
Actually - having it as a lemma is handy, since there isn't otherwise a `nextCoeff_zero` to just take `mt nextCoeff_zero`. I did golf it down to a proof term `ne_of_apply_ne nextCoeff h` which is not hard to write anywhere else. (I'm making this note partially for myself so that I can use `ne_of_apply_ne nextCoeff h` later; partially wondering if there should be a `nextCoeff_zero : nextCoeff (0:R[X]) = 0`.)